eventhandling/getSliEvent.go

```diff
@@ -45,31 +43,56 @@ func (eh GetSliEventHandler) HandleEvent() error {
 		return nil
 	}
 
-	// get shkeptncontext
-	keptnCtx, err := types.ToString(eh.event.Context.GetExtensions()["shkeptncontext"])
+	// send started event
+	_, err = eh.keptnHandler.SendTaskStartedEvent(eventData, utils.ServiceName)
+
 	if err != nil {
-		return fmt.Errorf("could not determine keptnContext of input event: %s", err.Error())
+		errMsg := fmt.Sprintf("Failed to send task started CloudEvent (%s), aborting...", err.Error())
+		log.Println(errMsg)
+		return err
```

Comment:
Why do we log AND return the error? Shouldn't wrapping the error work (assuming errors are logged in the main event handler loop)?

Reply:
I don't think the error returned here is logged anywhere at the moment.

https://github.com/keptn-contrib/prometheus-service/blob/ac86eed9e6be7423afb23b34b3401826bfc8acc3/main.go#L88-L90

We construct `errMsg` here for better troubleshooting experience. Argueably we could return that errMsg instead of err